include/surf/config.hpp

```diff
@@ -60,6 +60,8 @@ const std::string KEY_WEIGHTS = "weights";
 const std::string KEY_WEIGHTS_G = "weights_g";
 const std::string KEY_W_AND_P = "W_and_P";
 const std::string KEY_W_AND_P_G = "W_and_P_g";
+const std::string KEY_DOC_SPLITTERS = "doc_splitters";
```

Comment:
Looks like this is the same as KEY_DOCBORDER and KEY_DOCBORDER_RANK.
Maybe you can just use them.